### PR TITLE
[MM-56340] Add audit events for OAuth logins

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
While the ticket is only talking about the oauth logins, I've gone through the whole file and added audit events to all methods.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-56340

#### Release Note
```release-note
Add audit events for OAuth logins
```

### CODE REVIEW

File: server/channels/web/oauth.go [hunk 1]
```diff
@@ -76,7 +76,7 @@ func authorizeOAuthApp(c *Context, w http.ResponseWriter, r *http.Request) {
 	}
 
 	auditRec.Success()
-	c.LogAudit("")
+	c.LogAudit("success")
```
Comment (hanzei) [inline server/channels/web/oauth.go:79]: Unify style

Comment (wiggin77) [reply server/channels/web/oauth.go:79]: This new "style" is more error prone in that "success" can be misspelled.  Also the linter will eventually complain that the "success" string should be a constant.

Comment (wiggin77) [reply server/channels/web/oauth.go:79]: Why do we need two ways of marking an audit item as successful?

**Review by vish9812 — APPROVED** [review]

**Review by wiggin77 — APPROVED** [review]
Approved with non-blocking questions.

File: server/channels/web/oauth.go (hunk 1 above)
Comment (hanzei) [reply server/channels/web/oauth.go:79]: As far as I understand, there are two types of audit events in Mattermost:
1. The newly (by you) added audit logs, which use the advanced logging feature to write to, e.g. a file. (Created via `c.MakeAuditRecord`)
2. The (old) audit records, which are written to the DB and served via http://localhost:8065/admin_console/compliance/monitoring. (Compliance Monitoring) (Created via `c.LogAudit("success")`)

Based on the wording in the system console, Compliance Monitoring seems to be removed in a future release. Hence, I'm hesitant to invest more time into 2.

Does that clear things up @wiggin77 ?

**Comment by hanzei** [comment]
@DHaussermann The PR is ready for your review. You can find the QA test steps in the JIRA.

**Comment by DHaussermann** [comment]
@hanzei let's discuss this when we sync next. 

I see no change after deploying your code. Just the same 2 lines i the log.
`{"timestamp":"2024-03-11 14:15:12.946 -04:00","level":"debug","msg":"Received HTTP request","caller":"web/handlers.go:187","method":"POST","url":"/login/sso/saml","request_id":"wjecofyriiri3mghdyqp978y6c","user_id`

I think I missing some context on how to enable the auditing here.

**Comment by hanzei** [comment]
@DHaussermann Did you set up audit logs as https://docs.mattermost.com/manage/logging.html#audit-logging-beta?

**Comment by DHaussermann** [comment]
Hi @hanzei, I may need some more help on this. 
When I follow the steps as per the ticket I don't see any records getting added to the audit log.

- I **belive** have the auditing working as expected.
image

- I now have httpie working and when I send the invalid auth request I believe I'm seeing the expected response
image

- I'm not filtering out anything for logging
image
   
0/5 I guess if you're able to bring the branch current I can again. Please let me know if you have any other suggestions of something I can check.

**Comment by hanzei** [comment]
@DHaussermann Just FYI: The advanced logging field has nothing to do with the audit logs. It's for standard log messages.

**Comment by hanzei** [comment]
The audit logs look functional. Are the log events shown in the screenshot the latest ones? Can you please access the system console before running the command? That triggers an audit event and helps us identify if the adit logs work.

**Review by DHaussermann — APPROVED** [review]
Tested and passed
- Using the updated request in the Jira, I can see the auditing now occurs
- `{"timestamp":"2024-05-06 10:57:40.786 -04:00","event_name":"authorizeOAuthPage","status":"fail","actor":` 
- For this PR testing the failed auth path was far more significant the testing the successful auth past
- As a precaution, I will regression test oAuth login workflow post-merge if there is auditing setup to see rthe success path get logged as well.

LGTM!

Thanks for the help on this @hanzei !

**Pull request merged**